wdtk-client/src/main/java/org/wikidata/wdtk/client/Client.java

```diff
@@ -204,6 +220,45 @@ private void openActions() {
 	}
 
 	/**
+	 * Writes a report file including the results of the
+	 * {@link DumpProcessingAction#getReport()} methods. If there is no report
+	 * filename specified the reports will be logged.
+	 * 
+	 * @throws IOException
+	 */
+	void writeReport() throws IOException {
+		StringBuilder builder = new StringBuilder();
+		for (DumpProcessingAction action : this.clientConfiguration
+				.getActions()) {
+			if (this.clientConfiguration.getReportFilename() != null) {
+				builder.append(action.getActionName() + ": ");
+				builder.append(action.getReport());
+				builder.append(System.getProperty("line.separator"));
+			} else {
+				logger.info(action.getActionName() + ": " + action.getReport());
+			}
+		}
+		if (this.clientConfiguration.getReportFilename() != null) {
+			Path outputDirectory = Paths.get(
+					this.clientConfiguration.getReportFilename()).getParent();
+			if (outputDirectory == null) {
+				outputDirectory = Paths.get(".");
+			}
+			DirectoryManager dm = DirectoryManagerFactory
+					.createDirectoryManager(outputDirectory);
+			OutputStream out = dm.getOutputStreamForFile(Paths
+					.get(this.clientConfiguration.getReportFilename())
+					.getFileName().toString());
+
+			OutputStream bufferedFileOutputStream = new BufferedOutputStream(
+					out, 1024 * 1024 * 5);
```

Comment:
That's a big buffer for the three lines of report you normally have. Maybe just use the default.